src/Sfa.Das.Sas.Indexer.ApplicationServices/Provider/Services/ProviderIndexer.cs

```diff
@@ -138,12 +143,51 @@ private IEnumerable<CoreProvider> CreateProviders(ProviderSourceDto source)
                 provider.Name = ukrlpProvider?.ProviderName;
                 provider.Addresses = ukrlpProvider?.ProviderContact.Select(_ukrlpProviderMapper.MapAddress);
                 provider.Aliases = ukrlpProvider?.ProviderAliases;
-                
-                var byProvidersFiltered = source.AchievementRateProviders.Rates.Where(bp => bp.Ukprn == provider.Ukprn);
 
                 provider.IsEmployerProvider = source.EmployerProviders.Providers.Contains(provider.Ukprn.ToString());
                 provider.IsHigherEducationInstitute = source.HeiProviders.Providers.Contains(provider.Ukprn.ToString());
 
+                _providerDataService.SetLearnerSatisfactionRate(source.LearnerSatisfactionRates, provider);
+                _providerDataService.SetEmployerSatisfactionRate(source.EmployerSatisfactionRates, provider);
+
+                yield return provider;
+            }
+        }
+
+        private IEnumerable<CoreProvider> CreateApiProviders(ProviderSourceDto source)
+        {
+            foreach (var roatpProvider in source.RoatpProviders)
+            {
+                if (roatpProvider.ProviderType == ProviderType.SupportingProvider || !IsDateValid(roatpProvider)) continue;
+                var ukrlpProvider = source.UkrlpProvidersApi.MatchingProviderRecords.FirstOrDefault(x => x.UnitedKingdomProviderReferenceNumber == roatpProvider.Ukprn);
+
+                CoreProvider provider;
+
+                if (source.CourseDirectoryUkPrns.Contains(int.Parse(roatpProvider.Ukprn)))
+                {
+                    var courseDirectoryProvider = source.CourseDirectoryProviders.Providers.First(x => x.Ukprn == int.Parse(roatpProvider.Ukprn));
```

Comment:
I think you can put this in a variable rather than parsing it multiple times in the lambda expression.